### PR TITLE
[stdlib] Change C-style for loop to Swift-style for-in loop

### PR DESCRIPTION
**Pull request by johnlui**
All changes are in one commit now, and it has been fully tested with no "C-style for loop" compiler warnings. @moiseev #826

### CODE REVIEW

**Comment by johnlui** [comment]
also #742

**Comment by moiseev** [comment]
Thanks @johnlui :+1:

@nadavrot , you wanted to postpone merging this in (https://github.com/apple/swift/pull/742#issuecomment-166951126) are we good now?

File: stdlib/public/core/Character.swift [hunk 1]
```diff
@@ -120,7 +120,7 @@ public struct Character :
   @warn_unused_result
   static func _smallSize(value: UInt64) -> Int {
     var mask: UInt64 = 0xFF
-    for var i = 0; i < 8; i += 1 {
+    for i in 0..<8 {
```
Comment (ephemer) [inline stdlib/public/core/Character.swift]: Is there an official style guide regarding these loops?
I find it easier to read when it's in the style of `for i in 0 ..< 8` (with that spacing)

Comment (dabrahams) [inline stdlib/public/core/Character.swift]: There’s no official guide.  I will say that the standard library should remain internally consistent, I don’t see any compelling reason to change what we’re doing with range operator spacing.

Comment (ephemer) [inline stdlib/public/core/Character.swift]: Re: internal consistency, the C-style formatting is also spaced in the way I suggested (it's not `var i=0; i<8; i+=1`).

Comparing `0..<3` and `0 ..< 3`, I find the latter significantly easier to process in every case.
The decision ultimately doesn't affect my life in any way, but being an open forum, I wanted to put my vote in for readability.

**Comment by moiseev** [comment]
Nice catch, @hmhv!
@johnlui please fix the 2 instances of using `continue` to incremenent loop variable.
For the lack of better way of doing it, here are screenshots.
screen shot 2016-01-08 at 12 02 35 pm
screen shot 2016-01-08 at 12 03 43 pm

**Comment by johnlui** [comment]
@moiseev OK, I will make a new pull request~

**Comment by johnlui** [comment]
There is a bug-fixed PR #923 , so this will be closed.

**Pull request closed without merging**